src/configLoaders/ConfigLoader.ts

```diff
@@ -0,0 +1,3 @@
+export default interface ConfigLoader {
```

Comment:
Can you explain what a `ConfigLoader` is and why it is needed? Maybe add some references to Jest documentation or code?

Reply:
Well, this is not necessarily Jest related. In the code we use an implementation of `ConfigLoader` to return either a react config (generated via the `ReactConfigLoader`) or the config in the `package.json` (via the `DefaultConfigLoader`). We can add more config loaders for different kinds of projects in the future when the configuration is not present in the package.json.

Reply:
We can also add a `ConfigLoader` implementation for the Typescript flavour of react in the future. I made it this way so that we can add new loaders on the fly.

Reply:
That explanation makes sense. I'd like to see it in source code comments, so people don't have to refer to this PR to find out why the interface exists.

Reply:
That's fine, I'll add the comments this evening hopefully.